### DISCUSSION TITLE
Need identity support for Azure OpenAI for enterprise deployment

### DISCUSSION THREAD

**Discussion by georgehu0815** [open-webui/open-webui#6734] [General]
Eliminate plan password is critical for enterprise deployment. It will reduce the risk of attack.

We need support identity based access for Azure OpenAI:

Following is a example how to use identity in the agentic applications:

For GPT models:

from azure.identity import DefaultAzureCredential, get_bearer_token_provider
from openai import AzureOpenAI

AZURE_OPENAI_VERSION="2024-05-01-preview"
AZURE_OPENAI_DEPLOYMENT="gpt"
AZURE_OPENAI_ENDPOINT="https://***.openai.azure.com/"

token_provider = get_bearer_token_provider(
    DefaultAzureCredential(), "https://cognitiveservices.azure.com/.default"
)

client = AzureOpenAI(
    api_version=AZURE_OPENAI_VERSION, 
    azure_endpoint=AZURE_OPENAI_ENDPOINT, 
    azure_ad_token_provider=token_provider
)

def chat(message):
    print("=====agent received",message)
    messages=[
            {"role": "system", "content": "You are a helpful assistant."},
            {"role": "user", "content": message
             }
        ]
    response = client.chat.completions.create(
        model=AZURE_OPENAI_DEPLOYMENT,
        messages=messages
    )
    print(messages)
    return { "rt": response.choices[0].message.content }

print(chat("tell me a joke"))

For Dall.E 👍 

# Note: DALL-E 3 requires version 1.0.0 of the openai-python library or later
import os
from azure.identity import DefaultAzureCredential
from openai import AzureOpenAI
import json
import requests
import time

from azure.identity import DefaultAzureCredential, get_bearer_token_provider
from openai import AzureOpenAI

token_provider = get_bearer_token_provider(
    DefaultAzureCredential(), "https://cognitiveservices.azure.com/.default"
)
client = AzureOpenAI(
    api_version="2024-05-01-preview",
    azure_endpoint="https://***.cognitiveservices.azure.com/",
    azure_deployment="dall-e",
    azure_ad_token_provider=token_provider
)
USER_PROMPT_GOES_HERE="""
Generate a thumbnail for a Youtube video educational channel named "Building Agentic Applications for ERA of AI." 
"""
result = client.images.generate(
    model="dall-e-3", # the name of your DALL-E 3 deployment
    prompt=USER_PROMPT_GOES_HERE,
    n=1
)
image_url = json.loads(result.model_dump_json())['data'][0]['url']
print(f"Image URL: {image_url}")

timestr = time.strftime("%Y%m%d-%H%M%S")
img_data = requests.get(image_url).content
outfilename=f'agentic-app-{timestr}.jpg'
with open(outfilename, 'wb') as handler:
    handler.write(img_data)
print(f"Image saved as {outfilename}")

**Comment by Maleya** [comment]
I agree, there should be support to use DefaultAzureCredential by importing it as 
```from azure.identity import DefaultAzureCredential```
Enterprise level development is often done without keys (RBAC)
